Code/Legacy/CryCommon/IMovieSystem.h

```diff
@@ -479,22 +499,39 @@ struct IAnimTrack
     // Adds new keys if required.
     //////////////////////////////////////////////////////////////////////////
     virtual void SetValue(float time, const float& value, bool bDefault = false, bool applyMultiplier = false) = 0;
+    /**
+     * O3DE_DEPRECATION_NOTICE(GHI-9133)
+     * use equivalent SetValue that accepts AZ::Vector3  
+     **/
     virtual void SetValue(float time, const Vec3& value, bool bDefault = false, bool applyMultiplier = false) = 0;
+    /**
+     * O3DE_DEPRECATION_NOTICE(GHI-9133)
+     * use equivalent SetValue that accepts AZ::Vector4  
+     **/
     virtual void SetValue(float time, const Vec4& value, bool bDefault = false, bool applyMultiplier = false) = 0;
+    /**
+     * O3DE_DEPRECATION_NOTICE(GHI-9133)
+     * use equivalent SetValue that accepts AZ::Quaternion  
+     **/
     virtual void SetValue(float time, const Quat& value, bool bDefault = false) = 0;
     virtual void SetValue(float time, const bool& value, bool bDefault = false) = 0;
     virtual void SetValue(float time, const Maestro::AssetBlends<AZ::Data::AssetData>& value, bool bDefault = false) = 0;
 
     // support for AZ:: vector types - re-route to legacy types
-    void SetValue(float time, AZ::Vector3& value, bool bDefault = false, bool applyMultiplier = false)
+    virtual void SetValue(float time, AZ::Vector4& value, bool bDefault = false, bool applyMultiplier = false)
+    {
+        Vec4 vec4(value.GetX(), value.GetY(), value.GetZ(), value.GetW());
+        SetValue(time, vec4, bDefault, applyMultiplier);
+    }
+    virtual void SetValue(float time, AZ::Vector3& value, bool bDefault = false, bool applyMultiplier = false)
     {
         Vec3 vec3(value.GetX(), value.GetY(), value.GetZ());
         SetValue(time, vec3, bDefault, applyMultiplier);
     }
-    void SetValue(float time, AZ::Quaternion& value, [[maybe_unused]] bool bDefault = false)
+    virtual void SetValue(float time, AZ::Quaternion& value, bool bDefault = false)
```

Comment:
Ah okay I see understood, it might be worth adding a comment for this to clarify, and doing a quick test to check it works okay (I know overriding and overloading can cause some hairy problems in C++ in certain situations and isn't usually encouraged 😬)

Reply:
> I guess we should opt to just fix the API endpoints and swap everything over at once?

I think that makes the most sense yeah, I'd go for that :+1: (let me know when you need us to take another look)

Reply:
I removed the virtual flash from the method.